### PR TITLE
Make it easier to use as a lib

### PR DESCRIPTION
**Pull request by byzhang**
moving json from techempower to common;
making some char* const.

### CODE REVIEW

File: common/lwan.h [hunk 1]
```diff
@@ -304,6 +304,7 @@ int lwan_connection_get_fd(lwan_connection_t *conn)
     __attribute__((pure)) __attribute__((warn_unused_result));
 
 #if defined (__cplusplus)
+#include "lwan-serve-files.h"
```
Comment (lpereira) [inline common/lwan.h]: I wouldn't include this file here if using C++: let the user include both `lwan.h` and `lwan-serve-files.h`.

File: common/CMakeLists.txt [hunk 2]
```diff
@@ -2,6 +2,7 @@ set(SOURCES
 	base64.c
 	hash.c
 	int-to-str.c
+  json.c
```
Comment (lpereira) [inline common/CMakeLists.txt]: Indentation seems off here.

File: common/lwan.h (hunk 1 above)
Comment (byzhang) [inline common/lwan.h]: Fixed.

File: common/CMakeLists.txt (hunk 2 above)
Comment (byzhang) [inline common/CMakeLists.txt]: Fixed.

File: techempower/CMakeLists.txt [hunk 3]
```diff
@@ -21,7 +21,6 @@ if (MYSQL_INCLUDE_DIR AND SQLITE_FOUND)
 
 		add_executable(techempower
 			techempower.c
-			json.c
```
Comment (byzhang) [inline techempower/CMakeLists.txt]: Json decoding is pretty helpful to parse the POST data. And lwan is a high performance engine for REST services which also needs both JSON encoding and decoding.

**Comment by lpereira** [comment]
Sorry for taking so long to reply, I've been pretty busy with the template parser rewrite.

I'd make JSON as another library that your program can link to, at least when I've fully reviewed that code. I've picked that library for the sole purpose of making it to the Techempower benchmark. It's fast and has a sane API, but I need to understand all of it before it can go into the core. That's easy to do with CMake, though.

Also, please make different changes in different commits. Making a lot of unrelated changes in one single commit breaks things like git bisect, which are pretty useful to find regressions.

It's OK to force-push to the branch you've used to make the pull request, so using 'git rebase -i'  to squash/fixup/reorder commits is perfectly fine.

Cheers!

**Comment by byzhang** [comment]
updated.

**Comment by lpereira** [comment]
Merged, thank you!

**Pull request closed without merging**